### ISSUE TITLE
cmd/compile/internal/types2: bug in type inference

### ISSUE THREAD

**Issue by griesemer** [golang/go#45548]
For the following program
```Go
package p

func f[F interface{type *Q}, G interface{type *R}, Q, R any](q Q, r R) {}

func _() {
	f[*float64](1, 2)
}
```
type inference infers for the call of `f`:
```
### inferred targs = [*float64 *R₄ float64 int]
```
instead of
```
### inferred targs = [*float64 *int float64 int]
```
leading to an errror:
```
/Users/gri/tmp/test.go2:6:13: no error expected: "*R does not satisfy interface{type *R} (*R not found in *int)"
```

**Comment by griesemer** [comment]
cc: @findleyr

**Comment by ianlancetaylor** [comment]
Should types2 issues that only occur with generics be milestone Go1.17?  It doesn't seem like they need to be fixed for the 1.17 release.  Go1.18 might be a better milestone.

**Comment by griesemer** [comment]
I marked some of them that I hope to have fixed for 1.17 as such. This one I am actively working on and should be fixed by 1.17.
